plugins/woocommerce/includes/admin/class-wc-admin-addons.php

```diff
@@ -1556,4 +1544,55 @@ private static function set_locale_data_in_transient( $transient, $value, $local
 		$transient_value[ $locale ] = $value;
 		return set_transient( $transient, $transient_value, $expiration );
 	}
+
+	/**
+	 * Make wp_safe_remote_get request to Woo.com endpoint.
+	 * Optionally pass user auth token, locale or country.
+	 *
+	 * @param string $url
+	 * @param ?array $options     Options for the request. For example, to pass auth token,
+	 *                            locale and country:
+	 *                            array(
+	 *                            'auth' => true,
+	 *                            'locale' => true,
+	 *                            'country' => true,
+	 *                            )
+	 *
+	 * @return array|WP_Error
+	 */
+	public static function fetch( $url, $options = array() ) {
+		$headers = array();
+
+		if ( $options['auth'] ) {
```

Comment:
Before accessing any value from the `$options` array, let's ensure the data is available with `if ( ! empty($options['auth']) ) {`. Same for `locale`, `country` etc...